PhysicsTools/PatAlgos/plugins/PATMETProducer.cc

```diff
@@ -95,6 +105,18 @@ void PATMETProducer::produce(edm::Event & iEvent, const edm::EventSetup & iSetup
     // add the generated MET
     if (addGenMET_) amet.setGenMET((*genMETs)[idx]);
 
+
+    if (chsMETs.isValid()){
```

Comment:
this seems unsafe.
If variables should not be filled, the configuration should be set to not do it (e.g. empty labels).

Reply:
the issues is, we wanted to make the new track met floats a function of patMET to be called directly from miniaod (instead of user floats in miniaod, since it was discouraged). this is the same strategy as the calo met usage at the moment. however, unlike calo met, since the track met is not defined in reco/aod, the pat configurations fail. in the miniaod we now explicitly embed the new met producers for track met and chs met. if you have suggestions happy to work on it.

Reply:
If I understood correctly, this piece (if isValid) is addressing an issue in the PAT configuration which is not miniAOD.
- One option is to move pfMetCHS and pfMetTrk to the standard PAT configuration (why not? it appears dependent on PF candidates)
- the other option is to set chsmetSource to "" by default and make the code here ignore this input and set the appropriate values in the miniAOD_tools

Reply:
thank you for your suggestion, appreciate it a lot. We went ahead with the chsmetSource "" by default option, and edited the miniAOD tools accordingly to read the correct values.

Reply:
although now everything "works" and achieves the goals, this really is not pretty. After discussing with Matthieu, he suggested that we do: 

* (for reco / aod) add a new set of cfi files to reco met to produce these mets
* (for reco / aod) add them to the aod sequence
then for our current purpose
* remove the functions from pat met producers, add these as variations of MET (like caloMET)
* produce the pat chs met and pat track met collections in miniaod_tools
* and add the merging of the different pat mets in the PATMETSlimmer.cc

Is this plan acceptable from the point of view of the changes in the format (assuming we will comply with the time requirements)?

Reply:
I'm not sure I follow the proposal. If the idea is to add something to the RECO step so that it is in RECO/AOD, then we may have to wait for relvals to appear first to then make the next step and add relevant producers to miniAOD that will read these new products from RECO/AOD.

Reply:
propagating the changes on reco/aod to miniaod level is for the next time per say.. this does not have direct implications on the approach of adding it to pat met slimmer. although given we are working on this at the moment, we can add the reco component as well so we can re-optimize/minimize the code next time (eliminating the usage of EDProducer in miniaod tool)? but we can decouple the two discussions as well